yr/tutorials/Number_Guessing_Game.md

```diff
@@ -0,0 +1,512 @@
+
+Our new guessing game host Skill is now finished! Let's send it to Amazon and test it out.
+
+# Test Your Skill! 
+
+## Send the Code to Amazon
+
+At the bottom of the testing box, click on **Send Updates**. 
+
+![Send updates](../images/number_guessing_game/image33.png){:.enlargeImage}
+<p style="padding-bottom: 7px"></p>
+
+The updates should take about 1-2 minutes to finish updating. If you decide to make any changes to your Alexa Skill after completing this tutorial (e.g. updating/adding/deleting blocks), make sure you do this step again to send your new changes to Amazon! 
+
+## Test your Skill
+
+If you have an Alexa-enabled device handy, say *“Alexa, tell guessing game host to quiz me”* to it and hear the response. You can also do this on your mobile device by downloading the free **Amazon Alexa** app, which comes with a built-in Alexa simulator.
+
+If you don’t have an Alexa-enabled device on hand, the Testing page in App Inventor allows you to simulate an Alexa using your custom Alexa skill right in your browser! Simply type in the textbox at the bottom, *“tell guessing game host to quiz me”* (You can omit the Alexa at the beginning of the phrase when using the testing box). Does the conversation flow like we planned it to? Here’s an example conversation:
+
+Note that the "secret" number is randomly generated, so that the user will see different sequences than the example shown here.
+
+<p style="padding-bottom: 7px"></p>
+![Example Conversation](../images/number_guessing_game/image43.png){:.enlargeImage}
+<p style="padding-bottom: 7px"></p>
+
+## Finish!!
+
+Congratulations! You've finished programming the **Try Number tutorial** and learned about using **conditionals!**
+
+# Expand Your App
+
+Here are some ideas for ways to enhance your app! 
+
+The way our app is built as of now, once we guess the correct number, the game ends. Try changing the app so that you can "restart" the game once you guess the right number! 
+
+* Add a new line asking "Do you want to play again?" once the correct number is guessed.
+* Try implementing an utterance in your quizmeIntent such that when "Do you want to play again?" is asked and responded to accordingly, the game "resets" and a new secret number is set. 
+
+Currently, when given an incorrect guess, our app tells us that our guess is either too big or too small. Can you think of better hints that provide the player with more information?
+
+* For example, if the difference between the secret number and our guess is less than 10, have the app tell us: "Your guess is less than ten away from my secret number!" 
+* To do this, try implementing more conditionals that compare the guess with the secret number. Based on whether the comparison is true or not, have the app provide the appropriate response.
+
```

Comment:
The extensions that are natural to follow are:
1)some kind of tracking that tells the user at the end how many rounds it took them to find the secret number
2)reverse the role of Alexa and the user.  Let the user pick a random number and have Alexa find it by guessing and responding to "too low"  "too high" comments.  You may imply binary search method for Alexa of going to the middle number by updating the lowerBound and upperBound as needed.

Reply:
Hal said that reversing the role of Alexa and the user would be too complex for an extension

Reply:
We already have  Mobile CSP teachers working on this so Hal may not be aware of all our clientele.  So include it!